dhcpd/handler.go

```diff
@@ -130,11 +131,16 @@ func (server *Server) HandleMsg4(buf []byte, oob *ipv4.ControlMessage, peer net.
 
 	if req.IsOptionRequested(dhcpv4.OptionBootfileName) && !manifest.Suspended {
 		// serve iPXE script if user-class is iPXE, or whatever the user chooses if iPXE is disabled
-		if stringSlicesEqual(req.UserClass(), []string{"iPXE"}) || !manifest.Ipxe {
+		if stringSlicesEqual(req.UserClass(), []string{"iPXE"}) || !manifest.Ipxe.Enabled {
 			resp.Options.Update(dhcpv4.OptBootFileName(manifest.BootFilename))
 		} else if len(req.ClientArch()) > 0 && req.ClientArch()[0] > 0 {
 			// likely UEFI (not BIOS)
-			resp.Options.Update(dhcpv4.OptBootFileName("ipxe.efi"))
+			switch manifest.Ipxe.Type {
+			case mfest.RPi4:
+				resp.Options.Update(dhcpv4.OptBootFileName("ipxe_rpi4.efi"))
+			default:
+				resp.Options.Update(dhcpv4.OptBootFileName("ipxe.efi"))
+			}
```

Comment:
I haven't actually tried netbooting RPi 4, is there a hint in the DHCP request that would let us autodiscover it instead of requiring us to customize the manifest?